lib/ace/mode/haml_highlight_rules.js

```diff
@@ -11,116 +11,130 @@ var HamlHighlightRules = function() {
     // regexp must not have capturing parentheses. Use (?:) instead.
     // regexps are ordered -> the first match is used
     HtmlHighlightRules.call(this);
-    this.$rules.start.unshift(
-        {
-            token : "punctuation.section.comment",
-            regex : /^\s*\/.*/
-        },
-        {
-            token: "string.quoted.double",
-            regex: "==.+?=="
-        },
-        {
-            token: "keyword.other.doctype",
-            regex: "^!!!\\s*(?:[a-zA-Z0-9-_]+)?"
-        },
-        RubyExports.qString,
-        RubyExports.qqString,
-        RubyExports.tString,
-        {
-            token: "character.escape.haml",
-            regex: "^\\s*\\\\."
-        },
-        {
-            token: "text",
-            regex: /^\s*/,
-            next: "tag_single"
-        },
-        RubyExports.constantNumericHex,
-        RubyExports.constantNumericFloat,
-        
-        RubyExports.constantOtherSymbol,
-        {
-            token: "text",
-            regex: "=|-|~",
-            next: "embedded_ruby"
-        }
-    );
-    this.$rules.tag_single = [
-        {
-            token: "meta.tag.haml",
-            regex: /(%[\w:\-]+)/
-        },
-        {
-            token: "keyword.attribute-name.class.haml",
-            regex: "\\.[\\w-]+"
-        },
-        {
-            token: "keyword.attribute-name.id.haml",
-            regex: "#[\\w-]+"
-        },
-        {
-            token: "punctuation.section",
-            regex: "\\{",
-            next: "section"
-        },
-        
-        RubyExports.constantOtherSymbol,
-        
-        {
-            token: "text",
-            regex: /\s/,
-            next: "start"
-        },
-        {
-            token: "empty",
-            regex: "$|(?!\\.|#|\\{|\\[|=|-|~|\\/)",
-            next: "start"
-        }
-    ];
-    this.$rules.section = [
-        RubyExports.constantOtherSymbol,
-        
-        RubyExports.qString,
-        RubyExports.qqString,
-        RubyExports.tString,
-        
-        RubyExports.constantNumericHex,
-        RubyExports.constantNumericFloat,
-        {
-            token: "punctuation.section",
-            regex: "\\}",
-            next: "start"
-        } 
-    ];
-    
-    this.$rules.embedded_ruby = [ 
-        RubyExports.constantNumericHex,
-        RubyExports.constantNumericFloat,
-        {
+
+    this.$rules = {
+        "start": [
+            {
+                token: "comment.block", // multiline HTML comment
+                regex: /^\/$/,
+                next: "comment"
+            },
+            {
+                token: "comment.block", // multiline HAML comment
+                regex: /^\-#$/,
+                next: "comment"
+            },
+            {
+                token: "comment.line", // HTML comment
+                regex: /\/\s*.*/
+            },
+            {
+                token: "comment.line", // HAML comment
+                regex: /-#\s*.*/
```

Comment:
Is this the correct behavior of comments or just a workaround for missing indentation detection? On https://www.haml-converter.com/ all comments seem to be multiline. We can add indentation detection similar to https://github.com/ajaxorg/ace/blob/master/lib/ace/mode/yaml_highlight_rules.js#L74.